main.c

```diff
@@ -545,15 +554,22 @@ sexp run_main (int argc, char **argv) {
  done_options:
   if (!quit || main_symbol != NULL) {
     init_context();
+    load_init(i < argc || main_symbol != NULL);
     /* build argument list */
-    if (i < argc)
+    if (i < argc) {
+      args = SEXP_NULL;
       for (j=argc-1; j>=i; j--)
         args = sexp_cons(ctx, tmp=sexp_c_string(ctx,argv[j],-1), args);
-    /* if no script name, use interpreter name */
-    if (i >= argc || main_module != NULL)
-      args = sexp_cons(ctx, tmp=sexp_c_string(ctx,argv[0],-1), args);
-    load_init(i < argc || main_symbol != NULL);
-    sexp_set_parameter(ctx, sexp_meta_env(ctx), sym=sexp_intern(ctx, sexp_argv_symbol, -1), args);
+      sexp_set_parameter(
+        ctx, sexp_meta_env(ctx),
+        sym=sexp_intern(ctx, sexp_command_line_symbol, -1),
+        args);
+      sexp_env_define(
+        ctx, sexp_meta_env(ctx),
+        sym=sexp_intern(ctx, sexp_raw_script_file_symbol, -1),
+        sexp_c_string(ctx,argv[i],-1));
```

Comment:
Why are repeating these settings here and above?

Reply:
In `init_context` the two parameters are set to default values. Here they are set based on `argv[]`. If you can point at one central place where we could set them, that would be best.